module/VuFind/src/VuFind/Role/PermissionManager.php

```diff
@@ -90,6 +90,20 @@ public function isAuthorized($permission, $context = null)
         return false;
     }
 
+    /**
+     * Get a list of all configured permissions.
+     *
+     * @return string[]
+     */
+    public function getAllConfiguredPermissions(): array
+    {
+        $permissions = [];
+        foreach ($this->config as $value) {
+            $permissions = array_merge($permissions, (array)($value['permission'] ?? []));
+        }
+        return array_unique($permissions);
```

Comment:
Should this be wrapped in array_values() to reindex the array?

Reply:
I wavered back and forth on whether it was better to have a cleanly-indexed array or to save cycles by not doing extra reindexing. But it's probably more predictable to have a cleanly-indexed array, and it's not like this is being used in high-performance code, so I've added it as suggested.